modules/core/src/mathfuncs.cpp

```diff
@@ -1608,32 +1608,39 @@ void patchNaNs( InputOutputArray _a, double _val )
 
     Mat a = _a.getMat();
     const Mat* arrays[] = {&a, 0};
-    int* ptrs[1] = {};
+    int32_t* ptrs[1] = {};
     NAryMatIterator it(arrays, (uchar**)ptrs);
-    size_t len = it.size*a.channels();
```

Comment:
1. OK about pointers
2. The loop condition [should be](https://github.com/opencv/opencv/pull/23098#discussion_r1378086229) of a form `j < loop_invariant` and has to be calculated in signed integers to avoid unsigned wrapping over zero, so I propose to let `len` and `j` be ints rather than `size_t`